db/db.js

```diff
@@ -78,262 +47,275 @@ async function commitTransaction(client) {
+      'getAllSessionsFromLocation',
+      'SELECT * FROM sessions WHERE locationid = $1 order by start_time desc',
+      [location],
+      clientParam,
+    )
 
     if (!results) {
       return null
     }
 
-    return results.rows
-  } catch (e) {
-    helpers.log(`Error running the getAllSessionsFromLocation query: ${e}`)
-  }
-}
-
-// Gets the last session data from an unclosed session for a specified location
-async function getLastUnclosedSession(locationid) {
-  try {
-    const results = await pool.query(
-      "SELECT * FROM sessions WHERE locationid = $1  AND start_time > (CURRENT_TIMESTAMP - interval '7 days') AND end_time = null ORDER BY sessionid DESC LIMIT 1",
-      [locationid],
-    )
-    if (results === undefined) {
-      return null
-    }
-
-    return results.rows[0]
-  } catch (e) {
-    helpers.log(`Error running the getLastUnclosedSession query: ${e}`)
+    return results.rows.map(r => createSessionFromRow(r))
+  } catch (err) {
+    // Errors handled in runQuery
   }
 }
 
 // Creates a new session for a specific location
 async function createSession(phone, locationid, state, clientParam) {
   try {
-    let client = clientParam
-    const transactionMode = typeof client !== 'undefined'
-    if (!transactionMode) {
-      client = await pool.connect()
-    }
-    const results = await client.query('INSERT INTO sessions(phonenumber, locationid, state, od_flag) VALUES ($1, $2, $3, $4) RETURNING *', [
-      phone,
-      locationid,
-      state,
-      OD_FLAG_STATE.NO_OVERDOSE,
-    ])
-    if (!transactionMode) {
-      client.release()
-    }
+    const results = await runQuery(
+      'createSession',
+      'INSERT INTO sessions(phonenumber, locationid, state, od_flag) VALUES ($1, $2, $3, $4) RETURNING *',
+      [phone, locationid, state, OD_FLAG_STATE.NO_OVERDOSE],
+      clientParam,
+    )
     return results.rows[0]
-  } catch (e) {
-    helpers.log(`Error running the createSession query: ${e}`)
+  } catch (err) {
+    // Errors handled in runQuery
   }
 }
 
 // Enters the end time of a session when it closes and calculates the duration of the session
 async function updateSessionEndTime(sessionid, clientParam) {
   try {
-    let client = clientParam
-    const transactionMode = typeof client !== 'undefined'
-    if (!transactionMode) {
-      client = await pool.connect()
-    }
-
-    await client.query('UPDATE sessions SET end_time = CURRENT_TIMESTAMP WHERE sessionid = $1', [sessionid])
-    client.query("UPDATE sessions SET duration = TO_CHAR(age(end_time, start_time),'HH24:MI:SS') WHERE sessionid = $1", [sessionid]) // Sets the duration to the difference between the end and start time
-    if (!transactionMode) {
-      client.release()
-    }
-  } catch (e) {
-    helpers.log(`Error running the updateSessionEndTime query: ${e}`)
+    await runQuery('updateSessionEndTime', 'UPDATE sessions SET end_time = CURRENT_TIMESTAMP WHERE sessionid = $1', [sessionid], clientParam)
```

Comment:
this is really a lot nicer. I don't think even the try/catch is really a strike against it because we need to handle errors for the queries anyways so its not even a net addition. Quality improved 🌟